### PR TITLE
feat: Add async support to Bun.serve onError 

### PR DESCRIPTION
**Pull request by LukasKastern**
### What does this PR do?

Handle promise returned from onError.
Fixes: #8217 

- [x] Code changes

### How did you verify your code works?

I wrote automated tests

### CODE REVIEW

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -2961,6 +2966,75 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
...
+        fn processOnErrorPromise(
+            ctx: *RequestContext,
+            promise_js: JSC.JSValue,
+            promise: JSC.AnyPromise,
+            value: JSC.JSValue,
+            status: u16,
+        ) void {
+            var vm = ctx.server.vm;
+
+            switch (promise.status(vm.global.vm())) {
```
Comment (LukasKastern) [inline src/bun.js/api/server.zig:2978]: This is a copy of what `RequestContext.onResponse` is doing. I'm not sure if this is the correct way of handling the promise.

File: src/bun.js/api/server.zig [hunk 2]
```diff
@@ -2961,6 +2966,75 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
...
+                    ctx.finishRunningErrorHandler(promise.result(vm.global.vm()), status);
+                    return;
+                },
+            }
+
+            // Promise is not fulfilled yet
+            {
+                ctx.flags.is_error_promise_pending = true;
+                ctx.pending_promises_for_abort += 1;
+                promise_js.then(
```
Comment (LukasKastern) [inline src/bun.js/api/server.zig:3029]: I added new callbacks for the error resolve/reject but realized that using `onResolve` / `onError` is working fine due to the deduplication check in `runErrorHandlerWithStatusCodeDontCheckResponded`.

**Comment by TiBianMod** [comment]
Many many thanks again @LukasKastern

Vielen Dank 😄

**Review by cirospaciari — APPROVED** [review]
LGTM Thank you

**Pull request merged**